cabal.project

```diff
@@ -1,4 +1,4 @@
-index-state: 2021-02-15T00:00:00Z
```

Comment:
I don't know if this should be a blocker, but I have been assuming we get eg better caching by using the same `index-state` that our upstream dependency `cardano-ledger-specs` is using.

Do you have any clarity on what the pros/cons of using a different `index-state` are?

Reply:
I'm quite sure that this only influences the first time Hydra builds the stuff. Once that is done, the whole shell is cached and every dependency is also cached, so it shouldn't impact performance except in the first run.

It is true though that if we use the same as an upstream dependency they would have already built derivations that we will get for free, but it should benefit *only the first compilation*

Reply:
> it should benefit only the first compilation

I'm not sure when a build result gets added to Hydra. Is it only once it's merged? Or is it anytime a PR is built by CI? Etc.

For example: just me entering the `nix-shell` does not cause my resulting Nix closure to be uploaded to Hydra.

Reply:
I think Hydra has to build it on CI, and probably it only keeps artifacts that are created when merging into master.

Reply:
When Hydra evals and builds the project, it will create artifacts/derivations which will be cached for future builds (until eventual GC). 

It is true that if we don't have the same version of hackage index-state than our upstream dependencies, their derivations will most likely not work for us as they might use a different version of some dependencies, and we will build all of cardano-ledger-specs on our side with our index-state.

I'm running a build at their index-state to see if everything works as expected, and we can build HLS. If so, I will revert the change on the index-state.